libtransmission/announcer.c

```diff
@@ -153,6 +181,32 @@ typedef struct tr_announcer
 }
 tr_announcer;
 
+static struct tr_scrape_info* tr_announcerGetScrapeInfo(struct tr_announcer* announcer, const char* url)
+{
+    struct tr_scrape_info* info = NULL;
+
+    if (url && *url)
+    {
+        int pos;
+        bool found;
+        struct tr_scrape_info key;
+
+        key.url = (char*) url;
+        pos = tr_ptrArrayLowerBound(&announcer->scrape_info, &key, compareScrapeInfo, &found);
+        if (!found)
+        {
+            info = tr_new0(struct tr_scrape_info, 1);
+            info->url = tr_strdup(url);
+            info->multiscrape_max = TR_MULTISCRAPE_MAX;
+            tr_ptrArrayInsert(&announcer->scrape_info, info, pos);
+        }
+
+        info = tr_ptrArrayNth(&announcer->scrape_info, pos);
```

Comment:
No point in getting something we already have (since we just inserted it above):
```suggestion
        else
        {
            info = tr_ptrArrayNth(&announcer->scrape_info, pos);
        }
```

Reply:
Added cfa89c72f810633d880e8114edf655233b0ad08e